### PR TITLE
new package: conquest

### PR DESCRIPTION
**Pull request by tkoskela**
Create new package for DFT simulation code CONQUEST
https://github.com/OrderN/CONQUEST-release

### CODE REVIEW

File: var/spack/repos/builtin/packages/conquest/package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    homepage = "http://www.order-n.org/"
+    url = "https://github.com/OrderN/CONQUEST-release/archive/refs/tags/v1.2.tar.gz"
+    git = "https://github.com/OrderN/CONQUEST-release/"
+
+    # notify when the package is updated.
+    maintainers("davidbowler", "tkoskela", "ilectra")
+
+    version("1.2", sha256="74d974f20ec15ff31d97cd42aae6dbe95288eedfa785896d5872b9ff44ee7ae2")
+    version("1.1", sha256="772e058f073cccfee45521aa62bb13192ab07cb2979b6076ddbf21ba22f9ba5d")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/conquest/package.py:21]: Would it be possible to manually upload tarballs for these releases? That would ensure more stability of the sha256, see https://github.com/orgs/community/discussions/46034?sort=top for more details.

Comment (tkoskela) [reply var/spack/repos/builtin/packages/conquest/package.py:21]: That does certainly seem possible. Where should I manually upload them? The version tags were automatically created by `spack create`, is there a better way to add them?

Comment (alalazo) [reply var/spack/repos/builtin/packages/conquest/package.py:21]: Apologies, I have probably been too brief :slightly_smiling_face: I noticed the maintainers of the Spack package are also the developers of the package. Currently `conquest` relies on autogenerated tarballs from GitHub, but those don't grant stability of the shasum (se discussion linked above). 

I was wondering if it is fine with you to add an explicit artifact to each conquest release, and point the url in this package recipe to that newly uploaded artifact. Uploading should be possible when you edit the release in the Conquest repository. 

If, for any reason, you don't like the suggestion, let me know and I'll merge the package as is

Comment (tkoskela) [reply var/spack/repos/builtin/packages/conquest/package.py:21]: Thanks for the explanation! That shouldn't be a problem, but let me check with the main developer

Comment (tkoskela) [reply var/spack/repos/builtin/packages/conquest/package.py:21]: Should I keep the sha256 fields in the versions, or remove them?

Comment (alalazo) [reply var/spack/repos/builtin/packages/conquest/package.py:21]: You should keep them, and update the url and shasum.

Comment (alalazo) [reply var/spack/repos/builtin/packages/conquest/package.py:21]: Thanks!

File: var/spack/repos/builtin/packages/conquest/package.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    depends_on("blas")
+    depends_on("lapack")
+    depends_on("scalapack")
+    depends_on("fftw-api")
+    depends_on("libxc@:5")
+    depends_on("mpi")
+
+    variant("openmp", default=False, description="Build with OpenMP support")
+    variant("mult_kern", default="default", description="Matrix multiplication kernel type")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/conquest/package.py]: Can you add a list of allowed values?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/conquest/package.py (hunk 2 above)
Comment (tkoskela) [reply var/spack/repos/builtin/packages/conquest/package.py]: Done. I also had to fix a bug with the OpenMP variant, sorry about the extra change

**Review by alalazo — APPROVED** [review]

**Pull request merged**